Add OSP github action

julek-wolfssl opened the pull request.

julek-wolfssl committed:
Add OSP github action

julek-wolfssl commented:
Depends on https://github.com/wolfSSL/osp/pull/133

julek-wolfssl commented:
Now depends on https://github.com/julek-wolfssl/wolfssl-actions.

julek-wolfssl commented on .github/workflows/osp-check.yml:
```diff
@@ -0,0 +1,118 @@
+#  haproxy_check:
+#    strategy:
+#      matrix:
+#        os: [ ubuntu-latest ]
```
Removed.

julek-wolfssl commented on .github/workflows/osp-check.yml:
```diff
@@ -0,0 +1,118 @@
+  stunnel_check:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
```
Removed.

julek-wolfssl committed:
Code review

julek-wolfssl commented on .github/workflows/osp-check.yml:
```diff
@@ -0,0 +1,118 @@
+  openvpn_check:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
```
Removed.

julek-wolfssl committed:
Separate out into different files

julek-wolfssl commented on .github/workflows/main.yml:
```diff
@@ -10,11 +10,13 @@ on:
-    call_os-check:
+    os-check:
         uses: ./.github/workflows/os-check.yml
+    osp:
```
Please take a look again. Now the tests are in separate files.

bandi13 commented on .github/workflows/haproxy-check.yml:
```diff
@@ -0,0 +1,49 @@
+    strategy:
+      matrix:
+        # List of refs to test
+        ref: [ master ]
```
You don't actually use this variable.

julek-wolfssl committed:
Refactor os-check.yml

julek-wolfssl committed:
Actions fix

Fix for `workflows must be defined at the top level of the .github/workflows/ directory` error.

bandi13 reviewed (approved):
Looks good to me! Nice work.

bandi13 commented:
> Looks good to me! Nice work.

After the `make user_settings.h` test is fixed of course.

julek-wolfssl commented:
That should be the `user_settings.h` fixed.
